test/tutorials/lambda_theory.jl

```diff
@@ -0,0 +1,156 @@
+using Metatheory, Test, TermInterface
+
+# # Lambda theory
+#
+# This tutorial demonstrates how to implement a simple lambda calculus in Metatheory.
+# Importantly, it shows a practical example of [*e-graph analysis*](/egraphs/#EGraph-Analyses).
+# The three building blocks of lambda caluclus are *variables*, $\lambda$-functions, and *function
+# application*, which we can implement as subtypes of an abstract `LambdaExpr`ession:
+
```

Comment:
Can you please add a comment linking to https://michel.steuwer.info/files/publications/2024/EGRAPHS-2024.pdf presented at PLDI 24 at https://pldi24.sigplan.org/details/egraphs-2024-papers/10/Slotted-E-Graphs - saying that the best way of achieving this is through this new technique ?